tests/unit/conditions/test_address_allowlist_condition.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+from eth_account import Account
+
+from nucypher.policy.conditions.address import AddressAllowlistCondition
+from nucypher.policy.conditions.exceptions import (
+    InvalidCondition,
+    InvalidConditionContext,
+)
+from nucypher.policy.conditions.context import USER_ADDRESS_CONTEXT
+
+
+def test_address_allowlist_condition_init():
+    """Test the initialization of AddressAllowlistCondition."""
+    # Create test addresses
+    account1 = Account.create()
+    account2 = Account.create()
+
+    addresses = [account1.address, account2.address]
+
+    # Test successful initialization
+    condition = AddressAllowlistCondition(addresses=addresses)
+    assert condition.condition_type == "address-allowlist"
+    assert set(condition.addresses) == set(addresses)
+
+    # Test with empty addresses list
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(addresses=[])
+
+    # Test with invalid address
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(addresses=["not-an-ethereum-address"])
+
+    # Test with duplicate addresses
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(addresses=[account1.address, account1.address])
+
+
+def test_address_allowlist_condition_verify():
+    """Test the verification of AddressAllowlistCondition."""
+    # Create test accounts
+    allowed_account1 = Account.create()
+    allowed_account2 = Account.create()
+    not_allowed_account = Account.create()
+
+    # Create condition with allowed accounts
+    addresses = [allowed_account1.address, allowed_account2.address]
+    condition = AddressAllowlistCondition(addresses=addresses)
+
+    # Test successful verification with allowed account
+    context = {USER_ADDRESS_CONTEXT: {"address": allowed_account1.address}}
```

Comment:
This won't work if you are properly resolving the `:userAddress`. You can skip the implicit validation during resolution by doing something like the following - https://github.com/nucypher/nucypher/blob/main/tests/acceptance/conditions/test_conditions.py#L86 - see the patch at the top of the test where the resolving skips the validation and simply returns the expected address.